### ISSUE TITLE
Support for Templating in tile_layer_url for Dynamic Version Codes (Naver Maps in Korea)

### ISSUE THREAD

**Issue by hyungyunlim** [nathan-gs/ha-map-card#94]
Hello,

First of all, thank you for your excellent work on the custom map card!

I'm using the custom map card in Home Assistant to display maps from Naver Maps, a Korean map service provider. Naver Maps requires a version code in the tile_layer_url that changes periodically. To keep the map tiles up to date without manually updating the version code, I attempted to use a sensor in Home Assistant that fetches the latest version code and inserts it into the tile_layer_url dynamically.

However, it seems that the custom map card does not support templating or placeholders in the tile_layer_url field. When I use a template or a placeholder for the version code, the map does not display—only the device icons appear on a blank background.

This is how I did.

**1. Create a Sensor for the Version Code:**
```
sensor:
  - platform: command_line
    name: Naver Map Version
    command: >
      curl -s 'https://map.pstatic.net/nrb/styles/basic.json?fmt=jpg&mt=bg.ol.ts.ar.lko' |
      grep -o '"version":"[^"]*"' |
      head -1 |
      cut -d '"' -f4
    scan_interval: 3600  # Update every hour
```

**2. Create a Template Sensor for the Tile URL:**
```
template:
  - sensor:
      - name: "Naver Map URL"
        state: >-
          https://map.pstatic.net/nrb/styles/basic/{{ states('sensor.naver_map_version') }}/{z}/{x}/{y}@2x.png?mt=bg.ol.ts.ar.lko
```

**3. Configure the Custom Map Card Using the Sensor:**
```
type: custom:map-card
history_start: 16 hours ago
auto_fit: true
tile_layer_url: "{{ states('sensor.naver_map_url') }}"
tile_layer_attribution: naver.com
entities:
  - entity: device_tracker....
    gradual_opacity: 0.5
  - entity: device_tracker....
    gradual_opacity: 0.5
```

**Expected Behavior:**

- The map should display correctly using the dynamic tile_layer_url that includes the latest version code from sensor.naver_map_version.

**Actual Behavior:**
- The map does not display; only the device tracker icons are visible on a blank background.
- No errors appear in the browser console.
- Using a static URL with a hardcoded version code works as expected.

**Example of Working Static Configuration:**
`tile_layer_url: "https://map.pstatic.net/nrb/styles/basic/1732841753/{z}/{x}/{y}@2x.png?mt=bg.ol.ts.ar.lko"`

**Additional Information:**

**Attempted Solutions:**
  - Used config-template-card to inject the dynamic URL, but encountered syntax errors or the map did not display.
  - Tested the sensor values, and they are correctly fetching the latest version code and constructing the URL.
  - Confirmed that the placeholders {z}, {x}, {y} are correctly used and not URL-encoded.

**Environment Details:**
- Custom Map Card Version: v1.6.0
- No relevant errors appear in the Home Assistant logs.
- Browser console does not show any errors when the map fails to display.

**Question:**
- Is it possible to support templating or placeholders in the tile_layer_url field of the custom map card to allow dynamic insertion of values like the version code?
- If not currently supported, would it be possible to add this feature in a future update?

**Comment by nathan-gs** [comment]
@hyungyunlim Thank you for your very extensive suggestion! In #48 a similar ask was made, looking to merge both!

**Comment by hyungyunlim** [comment]
> @hyungyunlim Thank you for your very extensive suggestion! In #48 a similar ask was made, looking to merge both! Let's continue the discussion in #48!

Thanks!

**Comment by nathan-gs** [comment]
@hyungyunlim I've implemented this in [v1.9.0](https://github.com/nathan-gs/ha-map-card/releases/tag/v1.9.0), take a look at the docs for how it works.
